Helm mtoffl01 (#10947)
DataDog/documentation: content/en/agent/kubernetes/integrations.md

[1]: /integrations/consul/
[2]: /agent/guide/agent-commands/
{{% /tab %}}
{{% tab "Helm" %}}

> The values.yaml file contains a `confd` section to define both static and Autodiscovery integration checks. You can find inline examples in the [sample values.yaml][1]. Each key becomes a file in the agent's conf.d directory.

Reviewer: ```suggestion
The `values.yaml` file contains a `confd` section to define both static and Autodiscovery integration checks. You can find inline examples in the sample [values.yaml][1]. Each key becomes a file in the Agent's `conf.d` directory.

```

```yaml
  confd:
    <INTEGRATION_NAME>.yaml: |-
      ad_identifiers:
        - <INTEGRATION_AUTODISCOVERY_IDENTIFIER>
      init_config:
        - <INIT_CONFIG>
      instances:
        - <INSTANCES_CONFIG>
```
See the [Autodiscovery Container Identifiers][1] documentation for information on the `<INTEGRATION_AUTODISCOVERY_IDENTIFIER>`.

> Note that the Helm chart has two `confd` sections: one for agent checks, and a second for cluster checks. If you're using the Cluster Agent and you're looking to configure Autodiscovery for a cluster check, follow along with the [cluster check configuration example][2] and make sure to include `cluster_check: true`. See the [Cluster Check documentation][3] for more context. 

Reviewer: ```suggestion
**Note**: The Helm chart has two `confd` sections: one for Agent checks, and a second for cluster checks. If you are using the Cluster Agent and looking to configure Autodiscovery for a cluster check, follow the [cluster check configuration example][2] and make sure to include `cluster_check: true`. See the [Cluster Check documentation][3] for more context. 
```

Unlike auto-conf files, **key-value stores may use the short OR long image name as container identifiers**, e.g. `redis` OR `redis:latest`.

[1]: /agent/faq/template_variables/
{{% /tab %}}
{{% tab "Helm" %}}

> The following configuration defines the integration template for `redis` containers with a custom password parameter:

Reviewer: ```suggestion
The following configuration defines the integration template for Redis containers with a custom password parameter:
```

```yaml
  confd
    redisdb.yaml: |-
      ad_identifiers:
        - redis
      init_config:
      instances:
        - host: %%host%%
          port: 6379
          password: %%env_REDIS_PASSWORD%%
```

> As a result, the agent will contain a `redis.yaml` file with the above configuration in the `/confd` directory.

Reviewer: ```suggestion
As a result, the Agent now contains a `redis.yaml` file with the above configuration in the `/confd` directory.
```

> **Note**: The `"%%env_<ENV_VAR>%%"` template variable logic is used to avoid storing the password in plain text, hence the `REDIS_PASSWORD` environment variable must be passed to the Agent. See the [Autodiscovery template variable documentation][1].

Reviewer: ```suggestion
**Note**: The `"%%env_<ENV_VAR>%%"` template variable logic is used to avoid storing the password in plain text. Hence, you must pass the `REDIS_PASSWORD` environment variable to the Agent. See the [Autodiscovery template variable documentation][1].
```
